conformance/utils/roundtripper/roundtripper.go

```diff
@@ -191,6 +193,106 @@ func IsRedirect(statusCode int) bool {
 	return false
 }
 
+// CaptureTLSRoundTrip makes a request with the provided parameters and returns the
+// captured request and response from echoserver. An error will be returned if
+// there is an error running the function but not if an HTTP error status code
+// is received.
+func (d *DefaultRoundTripper) CaptureTLSRoundTrip(request Request, cPem, keyPem []byte, server string) (*CapturedRequest, *CapturedResponse, error) {
+	cReq := &CapturedRequest{}
+	client := http.DefaultClient
+
+	// Create a certificate from the provided cert and key
+	cert, err := tls.X509KeyPair(cPem, keyPem)
+	if err != nil {
+		return nil, nil, fmt.Errorf("unexpected error creating cert: %w", err)
+	}
+
+	// Add the provided cert as a trusted CA
+	certPool := x509.NewCertPool()
+	if !certPool.AppendCertsFromPEM(cPem) {
+		return nil, nil, fmt.Errorf("unexpected error adding trusted CA: %w", err)
+	}
+
+	if server == "" {
+		return nil, nil, fmt.Errorf("unexpected error, server name required for TLS")
+	}
+
+	// Create the Transport for this provided host, cert, and trusted CA
+	client.Transport = &http.Transport{
+		TLSClientConfig: &tls.Config{
```

Comment:
Not sure if this is strictly necessary, but in ingress conformance we also specify a VerifyPeerCertificate func:

https://github.com/kubernetes-sigs/ingress-controller-conformance/blob/7ee2f3e2d283acb203be862cacd593c71afb1311/test/http/http.go#L75-L87

Reply:
I don't think it is strictly necessary unless you have custom cert verification that you want done on the peer.  https://github.com/golang/go/issues/16363 seems to indicate that was VerifyPeerCertificate's intended purpose.  I would expect this to be overkill unless it was also in the implementations for which we are testing conformance.